File: examples/python/get_property.py
# Copyright 2025 NXP
#
# SPDX-License-Identifier: BSD-3-Clause
"""Read CurrentVersion tag from UART device, specified as the first argument to the script."""
import sys
from pymboot import McuBoot, PropertyTag

if len(sys.argv) < 2:
    print("specify a UART device as the first argument")
    exit(1)

device = sys.argv[1]

with McuBoot(device) as boot:
    print("seding GetProperty")
    response = boot.get_property(PropertyTag.CurrentVersion)
    print(f"Status code: {boot.status_code}")
    print(f"Status string: {boot.status_code_str}")
    print(f"Property value: {response}")

File: examples/get_property.rs
// Copyright 2025 NXP
//
// SPDX-License-Identifier: BSD-3-Clause
//! Read and parse [`PropertyTag::CurrentVersion`] tag from UART device, specified as the first CLI argument.
use mboot::{
    McuBoot,
    protocols::{ProtocolOpen, uart::UARTProtocol},
    tags::property::{PropertyTag, PropertyTagDiscriminants},
};

fn main() -> anyhow::Result<()> {
    let mut args = std::env::args().skip(1);
    let device = args.next().expect("first program parameter must be a device name");

    let mut boot = McuBoot::new(UARTProtocol::open(&device)?);
    println!("sending GetProperty");
    let response = boot.get_property(PropertyTagDiscriminants::CurrentVersion, 0)?;
    if let PropertyTag::CurrentVersion(version) = response.property {
        println!(
            "received version: {version}, minor number is {}, parsed from response words {:#X?} with status code {}",
            version.minor, response.response_words, response.status
        );
    }

    Ok(())
}
